Tests/SwiftSyntaxTest/SyntaxCollectionsTests.swift

```diff
@@ -14,136 +14,253 @@ import XCTest
 import SwiftSyntax
 import _SwiftSyntaxTestSupport
 
-fileprivate func integerLiteralElement(_ int: Int) -> ArrayElementSyntax {
+fileprivate func intElement(_ int: Int) -> ArrayElementSyntax {
   let literal = TokenSyntax.integerLiteral("\(int)")
   return ArrayElementSyntax(
-    expression: IntegerLiteralExprSyntax(literal: literal)
+    expression: IntegerLiteralExprSyntax(literal: literal),
+    trailingComma: .commaToken()
   )
 }
 
-public class SyntaxCollectionsTests: XCTestCase {
-
-  public func testAppendingElement() {
-    let arrayElementList = ArrayElementListSyntax([
-      integerLiteralElement(0)
-    ])
-
-    let newArrayElementList = ArrayElementListSyntax(arrayElementList + [integerLiteralElement(1)])
-    XCTAssert(newArrayElementList.kind.isSyntaxCollection)
-    XCTAssertEqual(newArrayElementList.count, 2)
-    XCTAssertNotNil(newArrayElementList.child(at: 1))
-    XCTAssert(!newArrayElementList.child(at: 1)!.kind.isSyntaxCollection)
-    XCTAssertEqual("\(newArrayElementList.child(at: 1)!)", "1")
+fileprivate func assertSyntaxCollectionManipulation(
```

Comment:
That’s why I am performing all the transformations on an `ArrayExprSyntax` instead of the actual collection type `ArrayElementListSyntax`. That way the substructure test checks that we still have the collection anchored in a parent.